frontend/src/state/AppReducer.tsx

```diff
@@ -67,7 +67,7 @@ export const appStateReducer = (state: AppState, action: Action): AppState => {
         const pendingDocumentIndex = currentPendingDocuments.findIndex(conv => conv.id === document.id);
         const uploadedDocumentIndex = currentUploadedDocuments.findIndex(conv => conv.id === document.id);
     
-        if (pendingDocumentIndex !== -1 && (document.status === DocumentStatusState.Indexed || document.status === DocumentStatusState.Failed || document.status === DocumentStatusState.PollingTimeout)) {
+        if (pendingDocumentIndex !== -1 && (document.status === DocumentStatusState.Indexed || document.status === DocumentStatusState.Failed || document.status === DocumentStatusState.PollingTimeout || document.status === DocumentStatusState.PiiDetected)) {
```

Comment:
Can we refactor this massive boolean into a function?

Reply:
All set